client/crash_report_database.cc

```diff
@@ -204,6 +206,69 @@ bool CrashReportDatabase::UploadReport::Initialize(const base::FilePath& path,
   return reader_->Open(path);
 }
 
+bool CrashReportDatabase::Envelope::Initialize(const base::FilePath& path) {
+  path_ = path;
+  if (path.empty()) {
+    return false;
+  }
+  writer_ = std::make_unique<FileWriter>();
+  if (!writer_->Open(
+          path, FileWriteMode::kReuseOrCreate, FilePermissions::kOwnerOnly)) {
+    return false;
+  }
+  writer_->Seek(0, SEEK_END);
+  return true;
+}
+
+CrashReportDatabase::Envelope::Envelope(const UUID& uuid) : uuid_(uuid) {}
+
+void CrashReportDatabase::Envelope::AddAttachments(
+    const std::vector<base::FilePath>& attachments) {
+  for (const auto& attachment : attachments) {
+    std::string contents;
+    base::FilePath basename = attachment.BaseName();
+    const std::vector<base::FilePath> kFilter = {
+#if BUILDFLAG(IS_WIN)
+        base::FilePath(L"__sentry-event"),
+#else
+        base::FilePath("__sentry-event"),
+#endif
+    };
+    if (std::find(kFilter.begin(), kFilter.end(), basename) != kFilter.end() ||
+        !LoggingReadEntireFile(attachment, &contents)) {
+      continue;
+    }
+
+    std::string header = base::StringPrintf(
+        "\n{\"type\": \"attachment\", "
+        "\"length\": %zu, "
+        "\"attachment_type\": \"event.attachment\", "
+        "\"filename\": \"%s\"}\n",
```

Comment:
We should probably escape filenames. Specifically, any special characters (they're valid in linux/macos filesystems but would produce invalid JSON) and backslash on Windows (valid separator but different meaning in JSON - this one is probably easiest to just replace with forward slash)